src/Parser/Warrior/Fury/Modules/Talents/FuriousSlashUptime.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatDuration, formatPercentage } from 'common/format';
+import { STATISTIC_ORDER } from 'Main/StatisticBox';
+import ExpandableStatisticBox from 'Main/ExpandableStatisticBox';
+import StatTracker from 'Parser/Core/Modules/StatTracker';
+import FuriousSlashTimesByStacks from '../Talents/FuriousSlashTimesByStacks';
+
+class FuriousSlashUptime extends Analyzer {
+	static dependencies = {
+    statTracker: StatTracker,
+    furiousSlashTimesByStacks: FuriousSlashTimesByStacks,
+  };
+  
+  get furiousSlashTimesByStack(){
+	  console.log(this.furiousSlashTimesByStacks.furiousSlashTimesByStacks);
+	  return this.furiousSlashTimesByStacks.furiousSlashTimesByStacks;
+  }
+  
+  get maxStackUptime(){
+	  const stacks = Object.values(this.furiousSlashTimesByStack).map((e, i) => e.reduce((a, b) => a + b, 0));
+	  return stacks[stacks.length - 1];
+	  //find the highest stack count possible, and return the uptime at that amount of stacks
+  }
+  
+  get uptimeSuggestionThresholds(){
+	  return{
+		  actual: (this.maxStackUptime / this.owner.fightDuration),
+		  isLessThan:{
+			  minor: 0.95,
+			  average: 0.9,
+			  major: 0.85,
+		  },
+		  style: 'percentage',
+	  };
+  }
+  
+  suggestions(when){
+	  const{
+		  isLessThan:{
+			  minor,
+			  average,
+			  major,
+		  },
+	  }=this.uptimeSuggestionThresholds;
+	  
+	  if(this.selectedCombatant.hasTalent(SPELLS.FURIOUS_SLASH_TALENT.id))
```

Comment:
Is this module useful when the talent is not selected? If not, the entire module should be deactivated if the talent wasn't selected, you can do this by setting the `active` prop in the constructor.

Reply:
This bit here still needs fixing. 
Seeing as you're already setting the active prop, you can simply remove the  if(this.selectedCombatant.hasTalent(SPELLS.FURIOUS_SLASH_TALENT.id)) { }

Reply:
Also the suggestions seems to follow our old way of doing them. 

You can simply do 
```suggestions(when) {
when(this.uptimeSuggestionsThresholds).addSuggestion(xxxx)
``` 

No need for the redundant adding of 
```when(this.maxStackUptime / this.owner.fightDuration).isLessThan(minor)```
or 
```
const{
	  isLessThan:{
    minor,
	  average,
	  major,
 },
 }=this.uptimeSuggestionThresholds;```